src/InterventionBackend.php

```diff
@@ -567,135 +475,95 @@ function (InterventionImage $resource) use ($width, $height) {
+                if ($useAsMinimum) {
+                    return $resource->scale($width, $height);
+                }
+                return $resource->scaleDown($width, $height);
             }
         );
     }
 
     /**
-     * Resize an image by width. Preserves aspect ratio.
-     *
-     * @param int $width
-     * @return static
+     * @inheritDoc
      */
-    public function resizeByWidth($width)
+    public function resizeByWidth(int $width): static
     {
         return $this->createCloneWithResource(
             function (InterventionImage $resource) use ($width) {
-                return $resource->widen($width);
+                return $resource->scale($width);
             }
         );
     }
 
     /**
-     * Resize an image by height. Preserves aspect ratio.
-     *
-     * @param int $height
-     * @return static
+     * @inheritDoc
      */
-    public function resizeByHeight($height)
+    public function resizeByHeight(int $height): static
     {
         return $this->createCloneWithResource(
             function (InterventionImage $resource) use ($height) {
-                return $resource->heighten($height);
+                return $resource->scale(height: $height);
             }
         );
     }
 
     /**
-     * Return a clone of this image resized, with space filled in with the given colour
-     *
-     * @param int $width
-     * @param int $height
-     * @param string $backgroundColor
-     * @param int $transparencyPercent
-     * @return static
+     * @inheritDoc
      */
-    public function paddedResize($width, $height, $backgroundColor = "FFFFFF", $transparencyPercent = 0)
+    public function paddedResize(string $width, string $height, string $backgroundColour = 'FFFFFF', int $transparencyPercent = 0): static
     {
         $resource = $this->getImageResource();
         if (!$resource) {
             return null;
         }
 
-        // caclulate the background colour
-        $background = $resource->getDriver()->parseColor($backgroundColor)->format('array');
-        // convert transparancy % to alpha
-        $background[3] = 1 - round(min(100, max(0, $transparencyPercent)) / 100, 2);
+        if ($transparencyPercent < 0 || $transparencyPercent > 100) {
+            throw new InvalidArgumentException('$transparencyPercent must be between 0 and 100. Got ' . $transparencyPercent);
+        }
+
+        $bgColour = Color::create($backgroundColour);
+        // The Color class is immutable, so we have to instantiate a new one to set the alpha channel.
+        // No need to do that if both the $backgroundColor and $transparencyPercent are 0.
+        if ($bgColour->channel(Alpha::class)->value() !== 0 && $transparencyPercent !== 0) {
+            $channels = $bgColour->channels();
+            $alpha = (int) round(255 * (1 - ($transparencyPercent * 0.01)));
+            $bgColour = new Color($channels[0]->value(), $channels[1]->value(), $channels[2]->value(), $alpha);
+        }
```

Comment:
As per the upgrade guide:
> It is no longer possible to pass color values as array

Passing as a `Color` object instead. Alpha now has to be between 0 and 255, not 0 and 1.